Compiler crash "checked decl cannot have error type" starting with the development snapshot build `swift-DEVELOPMENT-SNAPSHOT-2025-02-15-a`

### Description

The compiler crashes with this message:

```
checked decl cannot have error type
(accessor_decl decl_context=0x11f844848 range=[Repro.swift:2:58 - line:2:58] <anonymous @ 0x11f893798> interface_type="<Self where Self : TextContainerViewProtocol> (Self) -> () -> <<error type>>" access=internal captures=(<generic>  type=τ_0_0) get for="delegate"
  (parameter "self" decl_context=0x11f893790) result="<<error type>>" thrown_type="<null>"
  (parameter_list))
```

### Reproduction

Reduced test `Repro.swift`
```swift
protocol TextContainerViewProtocol {
    var delegate: any TextContainerViewDelegate<Self>? { get set}
}

protocol TextContainerViewDelegate<View> {
    associatedtype View: TextContainerViewProtocol
}
```

Command line
```
% ~/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-10-a.xctoolchain/usr/bin/swiftc Repro.swift
error: compile command failed due to signal 6 (use -v to see invocation)
checked decl cannot have error type
(accessor_decl decl_context=0x14583e248 range=[Repro.swift:2:58 - line:2:58] <anonymous @ 0x14588cf98> interface_type="<Self where Self : TextContainerViewProtocol> (Self) -> () -> <<error type>>" access=internal captures=(<generic>  type=τ_0_0) get for="delegate"
  (parameter "self" decl_context=0x14588cf90) result="<<error type>>" thrown_type="<null>"
  (parameter_list))
...
```

### Stack dump

```text
Stack dump:
0.	Program arguments: /Users/hiroshi/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-09-a.xctoolchain/usr/bin/swift-frontend -frontend -c -primary-file Repro.swift -target arm64-apple-macosx15.0 -Xllvm -aarch64-use-tbi -enable-objc-interop -color-diagnostics -empty-abi-descriptor -resource-dir /Users/hiroshi/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-09-a.xctoolchain/usr/lib/swift -no-auto-bridging-header-chaining -module-name Repro -in-process-plugin-server-path /Users/hiroshi/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-09-a.xctoolchain/usr/lib/swift/host/libSwiftInProcPluginServer.dylib -plugin-path /Users/hiroshi/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-09-a.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Users/hiroshi/Library/Developer/Toolchains/swift-DEVELOPMENT-SNAPSHOT-2025-03-09-a.xctoolchain/usr/local/lib/swift/host/plugins -o /var/folders/ww/kxcj3hm12kd92gcwr9t3y7k40000gq/T/TemporaryDirectory.Lj8k4y/Repro-1.o
1.	Apple Swift version 6.2-dev (LLVM 527e08406d53663, Swift 911fbb6d932c543)
2.	Compiling with effective version 5.10
3.	While walking into 'TextContainerViewProtocol' (at Repro.swift:1:1)
4.	While verifying AccessorDecl getter for delegate (at Repro.swift:2:9)
5.	While verifying FuncDecl getter for delegate (at Repro.swift:2:9)
6.	While verifying AbstractFunctionDecl getter for delegate (at Repro.swift:2:9)
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x00000001062e49f0 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x00000001062e311c llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x00000001062e5038 SignalHandler(int, __siginfo*, void*) + 296
3  libsystem_platform.dylib 0x00000001954a2de4 _sigtramp + 56
4  libsystem_pthread.dylib  0x000000019546bf70 pthread_kill + 288
5  libsystem_c.dylib        0x0000000195378908 abort + 128
6  swift-frontend           0x0000000106360948 (anonymous namespace)::Verifier::verifyParsed(swift::AssociatedTypeDecl*) (.cold.2) + 0
7  swift-frontend           0x00000001020f3dac (anonymous namespace)::Verifier::verifyChecked(swift::ValueDecl*) + 300
8  swift-frontend           0x00000001020f4b20 (anonymous namespace)::Verifier::verifyChecked(swift::AbstractFunctionDecl*) + 856
9  swift-frontend           0x00000001020f4ce0 (anonymous namespace)::Verifier::verifyChecked(swift::FuncDecl*) + 284
10 swift-frontend           0x00000001020e7460 (anonymous namespace)::Verifier::walkToDeclPost(swift::Decl*) + 1496
11 swift-frontend           0x00000001020f59d0 (anonymous namespace)::Traversal::doIt(swift::Decl*) + 308
12 swift-frontend           0x00000001020f8408 (anonymous namespace)::Traversal::visit(swift::Decl*) + 844
13 swift-frontend           0x00000001020f59b8 (anonymous namespace)::Traversal::doIt(swift::Decl*) + 284
14 swift-frontend           0x00000001020f7910 swift::Pattern* llvm::function_ref<swift::Pattern* (swift::Pattern*)>::callback_fn<(anonymous namespace)::Traversal::doIt(swift::Pattern*)::'lambda'(swift::Pattern*)>(long, swift::Pattern*) + 376
15 swift-frontend           0x00000001020f6e50 swift::Pattern* (anonymous namespace)::Traversal::traverse<swift::Pattern>(swift::ASTWalker::PreWalkResult<swift::Pattern*>, llvm::function_ref<swift::Pattern* (swift::Pattern*)>, llvm::function_ref<swift::ASTWalker::PostWalkResult<swift::Pattern*> (swift::Pattern*)>) + 84
16 swift-frontend           0x00000001020f7958 swift::Pattern* llvm::function_ref<swift::Pattern* (swift::Pattern*)>::callback_fn<(anonymous namespace)::Traversal::doIt(swift::Pattern*)::'lambda'(swift::Pattern*)>(long, swift::Pattern*) + 448
17 swift-frontend           0x00000001020f6e50 swift::Pattern* (anonymous namespace)::Traversal::traverse<swift::Pattern>(swift::ASTWalker::PreWalkResult<swift::Pattern*>, llvm::function_ref<swift::Pattern* (swift::Pattern*)>, llvm::function_ref<swift::ASTWalker::PostWalkResult<swift::Pattern*> (swift::Pattern*)>) + 84
18 swift-frontend           0x00000001020f8a2c (anonymous namespace)::Traversal::visit(swift::Decl*) + 2416
19 swift-frontend           0x00000001020f59b8 (anonymous namespace)::Traversal::doIt(swift::Decl*) + 284
20 swift-frontend           0x00000001020f8f94 (anonymous namespace)::Traversal::visitNominalTypeDecl(swift::NominalTypeDecl*) + 480
21 swift-frontend           0x00000001020f8148 (anonymous namespace)::Traversal::visit(swift::Decl*) + 140
22 swift-frontend           0x00000001020f59b8 (anonymous namespace)::Traversal::doIt(swift::Decl*) + 284
23 swift-frontend           0x00000001020f5890 swift::Decl::walk(swift::ASTWalker&) + 32
24 swift-frontend           0x00000001022a6190 swift::SourceFile::walk(swift::ASTWalker&) + 188
25 swift-frontend           0x00000001020e4874 swift::verify(swift::SourceFile&) + 96
26 swift-frontend           0x00000001023b0db4 swift::TypeCheckPrimaryFileRequest::cacheResult(std::__1::tuple<>) const + 76
27 swift-frontend           0x0000000101e78538 swift::performTypeChecking(swift::SourceFile&) + 92
28 swift-frontend           0x0000000100e605fc bool llvm::function_ref<bool (swift::SourceFile&)>::callback_fn<swift::CompilerInstance::performSema()::$_0>(long, swift::SourceFile&) + 16
29 swift-frontend           0x0000000100e57ac8 swift::CompilerInstance::forEachFileToTypeCheck(llvm::function_ref<bool (swift::SourceFile&)>) + 76
30 swift-frontend           0x0000000100e57a5c swift::CompilerInstance::performSema() + 76
31 swift-frontend           0x0000000100c2b334 withSemanticAnalysis(swift::CompilerInstance&, swift::FrontendObserver*, llvm::function_ref<bool (swift::CompilerInstance&)>, bool) + 60
32 swift-frontend           0x0000000100c20b98 performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 716
33 swift-frontend           0x0000000100c203b4 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 2244
34 swift-frontend           0x00000001009e2240 swift::mainEntry(int, char const**) + 3064
35 dyld                     0x00000001950ec274 start + 2840
```

### Expected behavior

No crash but this error

```
Repro.swift:2:23: error: use of protocol 'TextContainerViewDelegate<Self>' as a type must be written 'any TextContainerViewDelegate<Self>'
1 | protocol TextContainerViewProtocol {
2 |     var delegate: any TextContainerViewDelegate<Self>? { get set}
  |                       `- error: use of protocol 'TextContainerViewDelegate<Self>' as a type must be written 'any TextContainerViewDelegate<Self>'
3 | }
4 | 
```

### Environment

The latest macos development snapshot build https://download.swift.org/development/xcode/swift-DEVELOPMENT-SNAPSHOT-2025-03-10-a/swift-DEVELOPMENT-SNAPSHOT-2025-03-10-a-osx.pkg

A quick bisecting shows that this crash started with the build `swift-DEVELOPMENT-SNAPSHOT-2025-02-15-a` and doesn't reproduce with `swift-DEVELOPMENT-SNAPSHOT-2025-02-14-a`

### Additional information

_No response_

--- a/lib/Sema/TypeCheckType.cpp
+++ b/lib/Sema/TypeCheckType.cpp
@@ -915,22 +915,7 @@ static Type applyGenericArguments(Type type,
       argTys.push_back(argTy);
     }
 
-    auto parameterized =
-        ParameterizedProtocolType::get(ctx, protoType, argTys);
-
-    // FIXME: Can this not be done in ExistentialTypeSyntaxChecker?
-    if (resolution.getOptions().isConstraintImplicitExistential() &&
-        !ctx.LangOpts.hasFeature(Feature::ImplicitSome)) {
-      diags
-          .diagnose(loc, diag::existential_requires_any, parameterized,
-                    ExistentialType::get(parameterized),
-                    /*isAlias=*/isa<TypeAliasType>(type.getPointer()))
-          .warnUntilSwiftVersion(7);
-
-      return ErrorType::get(ctx);
-    }
-
-    return parameterized;
+    return ParameterizedProtocolType::get(ctx, protoType, argTys);
   }
   
   // Builtins have special handling.

--- a/test/type/explicit_existential.swift
+++ b/test/type/explicit_existential.swift
@@ -173,13 +173,13 @@ protocol Collection<T> {
 struct TestParameterizedProtocol<T> : Collection {
   typealias T = T
 
-  let x : Collection<T> // expected-warning {{use of protocol 'Collection<T>' as a type must be written 'any Collection<T>'}}
+  let x : Collection<T> // expected-warning {{use of protocol 'Collection' as a type must be written 'any Collection'}}
 }
 
 func acceptAny(_: Collection<Int>) {}
-// expected-warning@-1 {{use of protocol 'Collection<Int>' as a type must be written 'any Collection<Int>'}}
+// expected-warning@-1 {{use of protocol 'Collection' as a type must be written 'any Collection'}}
 func returnsAny() -> Collection<Int> {}
-// expected-warning@-1 {{use of protocol 'Collection<Int>' as a type must be written 'any Collection<Int>'}}
+// expected-warning@-1 {{use of protocol 'Collection' as a type must be written 'any Collection'}}
 
 func testInvalidAny() {
   struct S: HasAssoc {
@@ -374,8 +374,8 @@ func testAnyFixIt() {
   let _: Optional<HasAssoc>
   // expected-warning@+1 {{constraint that suppresses conformance requires 'any'}}{{19-28=any ~Copyable}}
   let _: Optional<~Copyable>
-  // FIXME: No fix-it + generic argument not diagnosed.
-  // expected-warning@+1 {{use of protocol 'HasAssocGeneric<any HasAssoc>' as a type must be written 'any HasAssocGeneric<any HasAssoc>'}}{{none}}
+  // expected-warning@+2:10 {{use of protocol 'HasAssocGeneric' as a type must be written 'any HasAssocGeneric'}}{{10-35=any HasAssocGeneric<HasAssoc>}}
+  // expected-warning@+1:26 {{use of protocol 'HasAssoc' as a type must be written 'any HasAssoc'}}{{26-34=any HasAssoc}}
   let _: HasAssocGeneric<HasAssoc>
   // expected-warning@+1 {{use of protocol 'HasAssoc' as a type must be written 'any HasAssoc'}}{{14-22=any HasAssoc}}
   let _: S.G<HasAssoc>
@@ -407,8 +407,8 @@ func testAnyFixIt() {
   // expected-warning@+2 {{use of protocol 'HasAssoc' as a type must be written 'any HasAssoc'}}{{13-21=any HasAssoc}}
   // expected-warning@+1 {{constraint that suppresses conformance requires 'any'}}{{10-37=any ~G<HasAssoc>.Copyable_Alias}}
   let _: ~G<HasAssoc>.Copyable_Alias
-  // FIXME: No fix-it + generic argument not diagnosed.
-  // expected-warning@+1 {{use of 'HasAssocGeneric<any HasAssoc>' as a type must be written 'any HasAssocGeneric<any HasAssoc>}}{{none}}
+  // expected-warning@+2:12 {{use of 'S.HasAssocGeneric_Alias' (aka 'HasAssocGeneric') as a type must be written 'any S.HasAssocGeneric_Alias' (aka 'any HasAssocGeneric')}} {{10-43=any S.HasAssocGeneric_Alias<HasAssoc>}}
+  // expected-warning@+1:34 {{use of protocol 'HasAssoc' as a type must be written 'any HasAssoc'}}{{34-42=any HasAssoc}}
   let _: S.HasAssocGeneric_Alias<HasAssoc>
   // FIXME: No diagnostic.
   let _: HasAssoc.Int_Alias
@@ -543,6 +543,8 @@ func testAnyFixIt() {
 
   // expected-error@+1 {{optional 'any' type must be written '(any HasAssoc)?'}}{{10-23=(any HasAssoc)?}}
   let _: any HasAssoc?
+  // expected-error@+1:10 {{optional 'any' type must be written '(any HasAssocGeneric<Int>)?'}}{{10-35=(any HasAssocGeneric<Int>)?}}
+  let _: any HasAssocGeneric<Int>?
   // FIXME: Better recovery
   // expected-error@+1 {{type '(any Copyable)?' cannot be suppressed}}
   let _: any ~Copyable?

--- a/test/type/parameterized_existential.swift
+++ b/test/type/parameterized_existential.swift
@@ -7,10 +7,10 @@ protocol Sequence<Element> { // expected-note {{'Sequence' declared here}}
 // 'any' is required here
 
 func takesSequenceOfInt1(_: Sequence<Int>) {}
-// expected-warning@-1 {{use of protocol 'Sequence<Int>' as a type must be written 'any Sequence<Int>'}}
+// expected-warning@-1 {{use of protocol 'Sequence' as a type must be written 'any Sequence'}}
 
 func returnsSequenceOfInt1() -> Sequence<Int> {}
-// expected-warning@-1 {{use of protocol 'Sequence<Int>' as a type must be written 'any Sequence<Int>'}}
+// expected-warning@-1 {{use of protocol 'Sequence' as a type must be written 'any Sequence'}}
 
 struct ConcreteSequence<Element> : Sequence {}
 
@@ -74,7 +74,7 @@ func saturation(_ dry: any Sponge, _ wet: any Sponge<Int, Int>) {
 
 func typeExpr() {
   _ = Sequence<Int>.self
-  // expected-warning@-1 {{use of protocol 'Sequence<Int>' as a type must be written 'any Sequence<Int>'}}
+  // expected-warning@-1 {{use of protocol 'Sequence' as a type must be written 'any Sequence'}}
 
   _ = any Sequence<Int>.self
   // expected-error@-1 {{'self' is not a member type of protocol 'parameterized_existential.Sequence<Swift.Int>'}}

